docs/site/HasMany-relation.md

```diff
@@ -0,0 +1,206 @@
+logic also designates the relation type and tries to infer the foreign key on
+the target model (`keyTo` in the relation metadata) to a default value (source
+model name appended with `id` in camel case, same as LoopBack 3). It also calls
+`property.array()` to ensure that the type of the property is inferred properly
+as an array of the target model instances.
+
+The decorated property name is used as the relation name and stored as part of
+the source model definition's relation metadata. The property type metadata is
+also preserved as an array of type `Order` as part of the decoration. Another
+usage of the decorator with a custom foreign key name for the above example is
+as follows:
+
+```ts
+// import statements
+class Customer extends Entity {
+  // constructor, properties, etc.
+  @hasMany(Order, {keyTo: 'custId'})
+  orders?: Order[];
+}
+```
+
+## Configuring a hasMany relation
+
+The configuration and resolution of a `hasMany` relation takes place at the
+repository level. Once `hasMany` relation is defined on the source model, then
+there are a couple of steps involved to configure it and use it. On the source
+repository, the following are required:
+
+- Use [Dependency Injection](Dependency-injection.md) to inject an instance of
+  the target repository in the constructor of your source repository class.
+- Declare a property with the factory function type
+  `HasManyRepositoryFactory<targetModel, typeof sourceModel.prototype.id>` on
+  the source repository class.
+- call the `_createHasManyRepositoryFactoryFor` function in the constructor of
+  the source repository class with the relation name (decorated relation
+  property on the source model) and target repository instance and assign it the
+  property mentioned above.
+
+The following code snippet shows how it would look like:
+
+{% include code-caption.html
+content="/src/repositories/customer.repository.ts.ts" %}
+
+```ts
+import {Order, Customer} from '../models';
+import {OrderRepository} from './order.repository.ts';
+import {
+  DefaultCrudRepository,
+  juggler,
+  HasManyRepositoryFactory,
+} from '@loopback/repository';
+
+class CustomerRepository extends DefaultCrudRepository<
+  Customer,
+  typeof Customer.prototype.id
+> {
+  public orders: HasManyRepositoryFactory<Order, typeof Customer.prototype.id>;
+  constructor(
+    @inject('datasources.db') protected db: juggler.DataSource,
+    @repository(OrderRepository) orderRepository: OrderRepository,
+  ) {
+    super(Customer, db);
+    this.orders = this._createHasManyRepositoryFactoryFor(
+      'orders',
+      orderRepository,
+    );
+  }
+}
+```
+
+The following CRUD APIs are now available in the constrained target repository
+factory `orders` for instances of `customerRepository`:
+
+- `create` for creating a target model instance belonging to customer model
+  instance
+  ([API Docs](https://apidocs.strongloop.com/@loopback%2fdocs/repository.html#DefaultHasManyEntityCrudRepository.prototype.create))
```

Comment:
I noticed that the apidocs for `HasManyInterface` is more descriptive. Should we replace the link with that anchor or nah